cmd/sonobuoy/app/splat.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2021 VMware Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/vmware-tanzu/sonobuoy/pkg/errlog"
+
+	"github.com/spf13/cobra"
+)
+
+func NewCmdSplat() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "splat AGGREGATOR_RESULTS_PATH",
+		Short: "Reads all tarballs in the specified directory and splat it to STDOUT (for internal use)",
+		Long:  "Sonobuoy is an introspective kubernetes component that generates reports on cluster conformance, configuration, and more",
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := runSplat(args[0]); err != nil {
+				errlog.LogError(err)
+				os.Exit(1)
+			}
+		},
+		Hidden: true,
+		Args:   cobra.ExactArgs(1),
+	}
+	return cmd
+}
+
+func runSplat(dirPath string) error {
+	sonobuoyResults, err := filepath.Glob(filepath.Join(dirPath, "*.tar.gz"))
+	if err != nil {
+		return err
+	}
+
+	// archive all the `files` found above into a single (in-memory) tarball
+	// meaning untar-ing `buffer` below would yield `files`, which are also tarballs
+	buf := bytes.NewBuffer([]byte{})
+
+	if err = loadResults(buf, sonobuoyResults); err != nil {
+		return err
+	}
+
+	fmt.Print(buf.String())
```

Comment:
So if I understand this, it tars the files into an in memory buffer then write that buffer to standard out  as string?

If so, we can make this a bit better.  Since we're using an streaming io, you really don't have to save it in buffer first. You should be able to stream the tar binary directly to stdout and voiding using a buffer.  This is important for large files specially.

Here is how this could work:
* change loadResult as follows 
  ```go
   func loadResult(stdout os.File, filenames []string){
       gzipWriter := gzip.NewWriter(stdout)
   }
   ```
* Call it as follow
  ```go
   if err = loadResults(os.Stdout, sonobuoyResults); err != nil {
		return err
   }
  ```

Reply:
Or something like that ^^^

Reply:
ah okay, my worry with that approach that it might get messy with open pipes to stream vs buffer. but maybe I'm overthinking it :smile: